airbyte-cdk/python/airbyte_cdk/sources/abstract_source.py

```diff
@@ -232,6 +235,8 @@ def _read_incremental(
         has_slices = False
         for _slice in slices:
             has_slices = True
+            if logger.isEnabledFor(logging.DEBUG):
+                yield AirbyteMessage(type=MessageType.LOG, log=AirbyteLogMessage(level=Level.INFO, message=f"slice:{json.dumps(_slice)}"))
             logger.debug("Processing stream slice", extra={"slice": _slice})
```

Comment:
open question: would it make sense to not using the logger here and only yield a the log message? I doubt there's a lot of value in logging both when running a read with the `--debug` flag
eg
```
{"type": "LOG", "log": {"level": "INFO", "message": "slice:{\"created_after\": \"1970-01-01T00:00:00.000Z\"}"}}
{"type": "DEBUG", "message": "Processing stream slice", "data": {"slice": "{'created_after': '1970-01-01T00:00:00.000Z'}"}}
```

Reply:
I had the exact same question. In the end, my conclusion was "I don't have the knowledge to know what was the use of this log". If it is operational, we would need to migrate the tooling associated to it. If we are confident it's not used, it will be my pleasure to remove it

Reply:
the debug flag is only used when manually troubleshooting commands (eg `python main.py read --config secrets/config.json --catalog integration_tests/configured_catalog.json --debug`)

Here's the [issue](https://github.com/airbytehq/airbyte/issues/3078) and it's corresponding [PR](https://github.com/airbytehq/airbyte/issues/3078) for extra context.

yielding an airbyte log message will be functionally equivalent since those messages will also show up in the terminal.

Reply:
Perfect! I'll remove the `logger.debug` lines then. Thanks!